tests/bgp/test_bgp_session.py

```diff
@@ -0,0 +1,160 @@
+
+    # verify sessions are established after test
+    if not duthost.check_bgp_session_state(bgp_neighbors):
+        local_interfaces = list(bgp_neighbors[bgp_neighbor]['interface'].keys())
+        for port in local_interfaces:
+            fanout, fanout_port = fanout_switch_port_lookup(fanouthosts, duthost.hostname, port)
+            if fanout and fanout_port:
+                logger.info("no shutdown fanout interface, fanout {} port {}".format(fanout, fanout_port))
+                fanout.no_shutdown(fanout_port)
+            neighbor_port = bgp_neighbors[bgp_neighbor]['interface'][port]['port']
+            neighbor_name = bgp_neighbors[bgp_neighbor]['name']
+            logger.info("no shutdown neighbor interface, neighbor {} port {}".format(neighbor_name, neighbor_port))
+            nbrhosts[neighbor_name]['host'].no_shutdown(neighbor_port)
+            time.sleep(1)
+
+        pytest_assert(wait_until(60, 10, 0, duthost.check_bgp_session_state, list(bgp_neighbors.keys())),
+                      "Not all BGP sessions are established on DUT")
+
+
+@pytest.mark.parametrize("test_type", ["bgp_docker", "swss_docker", "reboot"])
+@pytest.mark.parametrize("failure_type", ["interface", "neighbor"])
+@pytest.mark.disable_loganalyzer
+def test_bgp_session_interface_down(duthosts, rand_one_dut_hostname, fanouthosts, localhost,
+                                    nbrhosts, setup, test_type, failure_type):
+    '''
+    1: check all bgp sessions are up
+    2: inject failure, shutdown fanout physical interface or neighbor port or neighbor session
+    4: do the test, reset bgp or swss or do the reboot
+    5: Verify all bgp sessions are up
+    '''
+    duthost = duthosts[rand_one_dut_hostname]
+    neighbor = setup['test_neighbor']
+    neighbor_name = setup['neighhosts'][neighbor]['name']
+    local_interfaces = list(setup['neighhosts'][neighbor]['interface'].keys())
+
+    logger.debug("duthost {} neighbor {} interface {} test type {} inject failure type {}".format(
+        duthost, neighbor_name, local_interfaces, test_type, failure_type))
+
+    if failure_type == "interface":
+        for port in local_interfaces:
+            fanout, fanout_port = fanout_switch_port_lookup(fanouthosts, duthost.hostname, port)
+            if fanout and fanout_port:
+                logger.info("shutdown interface fanout {} port {}".format(fanout, fanout_port))
+                fanout.shutdown(fanout_port)
+                time.sleep(1)
+
+    elif failure_type == "neighbor":
+        logger.info("###### shutdown neighbor {}".format(failure_type))
+        for port in local_interfaces:
+            neighbor_port = setup['neighhosts'][neighbor]['interface'][port]['port']
+            logger.info("shutdown interface neighbor {} port {}".format(neighbor_name, neighbor_port))
+            nbrhosts[neighbor_name]['host'].shutdown(neighbor_port)
+            time.sleep(1)
+
+    duthost.shell('show ip bgp summary', module_ignore_errors=True)
```

Comment:
do we need to verify the bgp sessions are down here?

Reply:
Thanks @lolyu , will add a check here.